*: improve INL_HASH_JOIN document about collation incompatibility (#17119)
pingcap/docs: optimizer-hints.md

In this case, you need to place the hint directly after the `SELECT` keyword. For more details, see the [Syntax](#syntax) section.

> ### INL_JOIN, INL_HASH_JOIN, INL_MERGE_JOIN hint does not take effect due to collation incompatibility

Reviewer: ```suggestion
### INL_JOIN, INL_HASH_JOIN, and INL_MERGE_JOIN hints do not take effect due to collation incompatibility
```

> When the collation of the join key is incompatible between two tables, the `IndexJoin` operator cannot be utilized to execute the query. In this case, the [`INL_JOIN` hint](#inl_joint1_name--tl_name-), [`INL_HASH_JOIN` hint](#inl_hash_join) or [`INL_MERGE_JOIN`](#inl_merge_join) does not take effect. For example:

Reviewer: ```suggestion
When the collation of the join key is incompatible between two tables, the `IndexJoin` operator cannot be utilized to execute the query. In this case, the [`INL_JOIN`](#inl_joint1_name--tl_name-), [`INL_HASH_JOIN`](#inl_hash_join), and [`INL_MERGE_JOIN`](#inl_merge_join) hints do not take effect. For example:
```
